pkg/kubelet/envvars/envvars.go

```diff
@@ -34,16 +33,14 @@ func FromServices(services []*v1.Service) []v1.EnvVar {
 	for i := range services {
 		service := services[i]
 
-		// ignore services where ClusterIP is "None" or empty
-		// the services passed to this method should be pre-filtered
-		// only services that have the cluster IP set should be included here
-		if !v1helper.IsServiceIPSet(service) {
-			continue
-		}
-
 		// Host
 		name := makeEnvVariableName(service.Name) + "_SERVICE_HOST"
-		result = append(result, v1.EnvVar{Name: name, Value: service.Spec.ClusterIP})
+		if service.Spec.Type == v1.ServiceTypeExternalName {
+			result = append(result, v1.EnvVar{Name: name, Value: service.Spec.ExternalName})
```

Comment:
can an external name service also have a cluster IP? if so, should we prefer that?